pylearn2/datasets/hdf5.py

```diff
@@ -2,338 +2,472 @@
+        filename : str
+            HDF5 file name.
+        sources : list of str
+            A list of key(s) of dataset data in the HDF5 file.
+        spaces : list of Space objects
+            A list of spaces, one for each source in sources.
+        aliases : list of str, optional
+            A list of aliases, one for each source. They work as an alias for
+            the original name of the sources and once set can be used to access
+            the data. If you use this parameter, for each element in sources
+            you have to either specify an alias or None.
         load_all : bool, optional (default False)
-            If true, load dataset into memory.
+            If true, datasets are loaded into memory instead of being left
+            on disk.
+        cache_size: int, optional
+            This value is used when use_h5py is True.
+            The size in bytes for the chunk cache of the HDF5 library. Useful
+            when the HDF5 files has large chunks and when using a sequential
+            iterator. The chunk cache allows only access the disk for the
+            chunks and then copy the batches to GPU from memory, which can
+            result in a significant speed up. Sensible default values depend
+            on the size of your data and the batch size you wish to use. A
+            rule of thumb is to make a chunk contain 100 - 1000 batches and
+            make sure they encompass complete samples.
+        use_h5py: bool or 'auto', optional
+            Specifies if h5py or pytables should be used. If set to auto
+            pylearn2 will try to use pytables and will switch to h5py if
+            pytables cannot be loaded (e.g. is not installed).
+        kwargs : dict, optional
+            Keyword arguments passed to `DenseDesignMatrix`.
         """
+        assert isinstance(filename, basestring)
+        assert isinstance(sources, list)
+        assert all([isinstance(el, basestring) for el in sources])
+        assert isinstance(spaces, list)
+        assert all([isinstance(el, Space) for el in spaces])
+        assert len(sources) == len(spaces)
+        if aliases is not None:
+            assert isinstance(aliases, list)
+            assert all([isinstance(el, basestring) for el in aliases
+                       if el is not None])
+            assert len(aliases) == len(sources)
+        assert isinstance(load_all, bool)
+        assert cache_size is None or isinstance(cache_size, py_integer_types)
+        assert isinstance(use_h5py, bool) or use_h5py == 'auto'
+
+        self.load_all = load_all
+        if aliases is None:
+            aliases = ()
+            # set aliases to canonical name or to source name
+            for s in sources:
+                if s in ['X', 'topo_view']:
+                    aliases += ('features',)
+                elif s == 'y':
+                    aliases += ('targets',)
+                else:
+                    aliases += (s,)
```

Comment:
I still think the aliases behaviour can be useful, especially if you define an experiment with some names in your model and costs, and then want to use a different hdf5 file where datasets have been named differently.